flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -851,6 +837,20 @@ private boolean checkFilterCondition(
 				}
 			};
 		}
+
+		@Override
+		public Long timestamp() {
+			if (timerService.isProcessingTime()) {
+				return null;
```

Comment:
Should this return `null` in the case of processing time or the current processing time? Null seems a bit odd.

Reply:
This is to comply with the behavior in `ProcessFunction.Context`. We did it the same way for `org.apache.flink.cep.functions.PatternProcessFunction`.